unit-test/background/reference-tests/broken-site-reporting-tests.js

```diff
@@ -20,6 +20,13 @@ async function submitAndValidateReport (report) {
     })
     tab.upgradedHttps = report.wasUpgraded
 
+    if (report.protectionsEnabled === true) {
+        spyOnProperty(tab.site, 'enabledFeatures').and.returnValue(['contentBlocking'])
```

Comment:
sorry, I'm not familiar with this value - why when protections are enabled only "contentBlocking" is in "enabledFeatures"? And similarly, why, when site is denylisted, no protections are enabled?

Reply:
The spied/mocked properties here cover *just* enough code paths to give the desired outcome.

for example, when we want to mock a user being 'protected', setting enabledFeatures: ["contentBlocking"] is enough - because when you follow the code through, it checks the following things:

- is the site denylisted?
- is the site allowlisted?
- is the site 'broken' (meaning, it was an entry in unprotectedTemporary)?
- is the feature enabled for this site?

Perhaps a comment above these lines would help to clear up such questions?

Reply:
The same is true for denylisted - because of the checks that are made, if this property is true it causes an early return, so there's no need to mock anything else :)

Reply:
Gotcha, thanks for explanation.